src/decode/lzma.rs

```diff
@@ -188,33 +170,9 @@ where
 }
 
 // Initialize decoder with circular buffer
-pub fn new_circular<W>(
-    output: W,
-    params: LzmaParams,
-) -> error::Result<DecoderState<W, lzbuffer::LzCircularBuffer<W>>>
-where
-    W: io::Write,
-{
-    new_circular_with_memlimit(output, params, std::usize::MAX)
-}
-
-// Initialize decoder with circular buffer
-pub fn new_circular_with_memlimit<W>(
-    output: W,
-    params: LzmaParams,
-    memlimit: usize,
-) -> error::Result<DecoderState<W, lzbuffer::LzCircularBuffer<W>>>
-where
-    W: io::Write,
-{
+pub fn new_circular(params: LzmaParams) -> error::Result<DecoderState> {
```

Comment:
With the `output` extracted out, this becomes identical to `new_accum`. Can you merge the two into a single function called `new`?

Reply:
I've renamed this to `new` and moved it to an associated function of `DecoderState`